templates/nextflow-s3-template.yaml

```
@@ -23,18 +23,21 @@ Parameters:
     Description: (Optional) Name of the created bucket.
   ReadWriteAccessArns:
     Type: CommaDelimitedList
-    # Default: "[]"
-    Description: (Required) Grant read/write access to accounts, groups, and users.
+    Default: "[]"
+    Description: (Optional) Grant read/write access to accounts, groups, and users.
     ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
   ReadOnlyAccessArns:
     Type: CommaDelimitedList
-    # Default: "[]"
-    Description: (Required) Grant read access to accounts, groups, and users.
+    Default: "[]"
+    Description: (Optional) Grant read access to accounts, groups, and users.
     ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
 
 # Conditions
 Conditions:
   HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+  HasReadWriteAccessArns: !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], "[]"]]
+  HasReadOnlyAccessArns: !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], "[]"]]
+  NoReadWriteAccessArns: !Equals [!Join [",", !Ref ReadWriteAccessArns], "[]"]
```

Out of curiosity, can you refer to prior conditions in later conditions? If so, we could avoid some duplication using something like this: 

```suggestion
  NoReadWriteAccessArns: !Equals [!Join [",", !Ref ReadWriteAccessArns], "[]"]
  HasReadOnlyAccessArns: !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], "[]"]]
  HasReadWriteAccessArns: !Not [!Ref NoReadWriteAccessArns]
```

---

This change wouldn't be required if we dropped the `NoReadWriteAccessArns` condition as per my comment below.